logging/troubleshooting/cluster-logging-troubleshooting-for-critical-alerts.adoc

```diff
@@ -0,0 +1,536 @@
+.Troubleshooting
+
+Consider https://www.elastic.co/guide/en/elasticsearch/reference/current/important-settings.html#heap-size-settings[increasing the heap size].
+
+[id="aggregated-logging-system-cpu-is-high"]
+== Aggregated Logging System CPU is High
+
+System CPU usage on the node is high.
+
+.Troubleshooting
+
+Check the CPU of the cluster node. Consider allocating more CPU resources to the node.
+
+[id="elasticsearch-process-cpu-is-high"]
+== Elasticsearch Process CPU is High
+
+Elasticsearch process CPU usage on the node is high.
+
+.Troubleshooting
+
+Check the CPU of the cluster node. Consider allocating more CPU resources to the node.
+
+[id="elasticsearch-disk-space-is-running-low"]
+== Elasticsearch Disk Space is Running Low
+
+The Elasticsearch Cluster is predicted to be out of disk space within the next 6 hours based on current disk usage.
+
+.Troubleshooting
+
+. Get the disk space of the Elasticsearch node.
++
+[source,terminal]
+----
+for pod in `oc -n openshift-logging get po -l component=elasticsearch -o jsonpath='{.items[*].metadata.name}'`; do echo $pod; oc -n openshift-logging exec -c elasticsearch $pod -- df -h /elasticsearch/persistent; done
+----
+
+. In the command output, check the `nodes.node_name.fs` field to determine the free disk space on that node.
+
+. Try to increase the disk space on all nodes.
+
+. If increasing the disk space is not possible, try adding a new data node to the cluster.
+
+. If adding a new data node is problematic, decrease the total cluster redundancy policy.
+
+.. Check the current `redundancyPolicy`.
++
+[source,terminal]
+----
+oc -n openshift-logging get es elasticsearch -o jsonpath='{.spec.redundancyPolicy}'
+----
++
+NOTE: If you are using a `ClusterLogging` CR, enter:
++
+[source,terminal]
+----
+oc -n openshift-logging get cl -o jsonpath='{.items[*].spec.logStore.elasticsearch.redundancyPolicy}'
+----
+
+.. If the cluster `redundancyPolicy` is higher than `SingleRedundancy`, set it to `SingleRedundancy` and save this change.
+
+. If the preceding steps do not fix the issue, delete the old indices.
+
+.. Check the status of all indices on Elasticsearch.
++
+[source,terminal]
+----
+oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- indices
+----
+
+.. Identify an old index that can be deleted.
+
+.. Delete the index.
++
+[source,terminal]
+----
+oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=<elasticsearch_index_name> -X DELETE
+----
+
+
+.Additional resources
+
+* "redundancyPolicy: SingleRedundancy" in the "Sample `ClusterLogging` custom resource (CR)" in xref:../../logging/config/cluster-logging-configuring-cr.adoc#cluster-logging-configuring-crd_cluster-logging-configuring-cr[About the Cluster Logging custom resource]
+
+* "ElasticsearchDiskSpaceRunningLow" in xref:../../logging/troubleshooting/cluster-logging-alerts.adoc#cluster-logging-elasticsearch-rules_cluster-logging-alerts[About Elasticsearch alerting rules].
```

Comment:
Again, xref works, but the links from the "elasticsearch alerting rules" don't work. I click the links, and I get a 404 page. So I don't know how effective it is to take users to "elasticsearch alerting rules" because that page only leads to dead links.